File: spine-libgdx/src/com/esotericsoftware/spine/attachments/TextureAtlasAttachmentLoader.java

package com.esotericsoftware.spine.attachments;

import com.esotericsoftware.spine.Attachment;
import com.esotericsoftware.spine.AttachmentLoader;
import com.esotericsoftware.spine.AttachmentType;

import com.badlogic.gdx.graphics.g2d.TextureAtlas;
import com.badlogic.gdx.graphics.g2d.TextureAtlas.AtlasRegion;

public class TextureAtlasAttachmentLoader implements AttachmentLoader {
	private TextureAtlas atlas;

	public TextureAtlasAttachmentLoader (TextureAtlas atlas) {
		if (atlas == null) throw new IllegalArgumentException("atlas cannot be null.");
		this.atlas = atlas;
	}

	public Attachment newAttachment (AttachmentType type, String name) {
		Attachment attachment = null;
		switch (type) {
		case region:
			attachment = new RegionAttachment(name);
			break;
		case regionSequence:
			attachment = new RegionAttachment(name);
			break;
		default:
			throw new IllegalArgumentException("Unknown attachment type: " + type);
		}

		if (attachment instanceof RegionAttachment) {
			AtlasRegion region = atlas.findRegion(attachment.getName());
			if (region == null)
				throw new RuntimeException("Region not found in atlas: " + attachment + " (" + type + " attachment: " + name + ")");
			((RegionAttachment)attachment).setRegion(region);
		}

		return attachment;
	}
}

File: spine-ts/webgl/src/TextureAtlasAttachmentLoader.ts
module spine.webgl {
	export class TextureAtlasAttachmentLoader implements AttachmentLoader {
		atlas: TextureAtlas;

		constructor (atlas: TextureAtlas) {
			this.atlas = atlas;
		}

		/** @return May be null to not load an attachment. */
		newRegionAttachment (skin: Skin, name: string, path: string): RegionAttachment {
			let region = this.atlas.findRegion(path);
			region.renderObject = region;
		    if (region == null) throw new Error("Region not found in atlas: " + path + " (region attachment: " + name + ")");
		    let attachment = new RegionAttachment(name);
			attachment.setRegion(region);
		    attachment.region = region;
		    return attachment;
		}

		/** @return May be null to not load an attachment. */
		newMeshAttachment (skin: Skin, name: string, path: string) : MeshAttachment {
			let region = this.atlas.findRegion(path);
			region.renderObject = region;
		    if (region == null) throw new Error("Region not found in atlas: " + path + " (mesh attachment: " + name + ")");
		    let attachment = new MeshAttachment(name);            
		    attachment.region = region;
		    return attachment;
		} 

		/** @return May be null to not load an attachment. */
		newBoundingBoxAttachment (skin: Skin, name: string) : BoundingBoxAttachment {
			return new BoundingBoxAttachment(name);
		} 
		
		/** @return May be null to not load an attachment */
		newPathAttachment(skin: Skin, name: string): PathAttachment {
			return new PathAttachment(name);
		}  
	} 
}
